models/Asset.php

```diff
@@ -2036,20 +2025,13 @@ private function updateChildPaths(FilesystemOperator $storage, string $oldPath)
      */
     private function relocateThumbnails(string $oldPath)
     {
-        $oldParent = dirname($oldPath);
-        $newParent = dirname($this->getRealFullPath());
+        $oldParent = dirname($oldPath).'/'.$this->getId();
+        $newParent = dirname($this->getRealFullPath()).'/'.$this->getId();
         $storage = Storage::get('thumbnail');
 
         try {
-            //remove source parent folder thumbnails
-            $contents = $storage->listContents($oldParent)->filter(fn (StorageAttributes $attributes) => ($attributes->isFile() && strstr($attributes['path'], 'image-thumb_')));
-            /** @var StorageAttributes $item */
-            foreach ($contents as $item) {
-                $storage->delete($item['path']);
-            }
-
             //remove destination parent folder thumbnails
-            $contents = $storage->listContents($newParent)->filter(fn (StorageAttributes $attributes) => ($attributes->isFile() && strstr($attributes['path'], 'image-thumb_')));
+            $contents = $storage->listContents($newParent)->filter(fn (StorageAttributes $attributes) => str_contains($attributes['path'], 'image-thumb_'));
             /** @var StorageAttributes $item */
             foreach ($contents as $item) {
                 $storage->delete($item['path']);
```

Comment:
I'm afraid that this is still not entirely right 🤔
What we need to do is, deleting all the folder previews (generated [here](https://github.com/pimcore/pimcore/blob/486a9f9f3611c7159b89554d0ab8bf900a79e673/models/Asset/Folder.php#L115)) of the parent folders of the source and the target folder. 

So I think we could implement a `private` helper function, where we pass an instance of `Asset/Folder` and then this function goes up the tree on by one and deletes the preview images. 

Example: 
```php
    private function clearFolderThumbnailsOfParents(Asset $asset): void
    {
        while ($asset = $asset->getParent()) {
            $asset->clearThumbnails(true);
        }
    }
```

Reply:
Ah you mean because the just moved thumbnails may be inside the folder preview? Now I understand what this is good for ;-)

Reply:
Ok, changed it. What do you think now?